File: frontend/lib/models/flow_guide.dart
class WeatherInfo {
  final double temp;
  final double feelsLike;
  final String condition;
  final int humidity;
  final String city;

  const WeatherInfo({
    required this.temp,
    required this.feelsLike,
    required this.condition,
    required this.humidity,
    required this.city,
  });

  factory WeatherInfo.fromJson(Map<String, dynamic> j) => WeatherInfo(
        temp: (j['temp'] as num).toDouble(),
        feelsLike: (j['feels_like'] as num).toDouble(),
        condition: j['condition'] as String,
        humidity: j['humidity'] as int,
        city: j['city'] as String,
      );
}

class FlowGuideCard {
  final String phase;
  final String greeting;
  final String oneAction;
  final String actionReason;
  final int durationMinutes;
  final int? routineId;
  final bool outfitReady;

  const FlowGuideCard({
    required this.phase,
    required this.greeting,
    required this.oneAction,
    required this.actionReason,
    required this.durationMinutes,
    this.routineId,
    required this.outfitReady,
  });

  factory FlowGuideCard.fromJson(Map<String, dynamic> j) => FlowGuideCard(
        phase: j['phase'] as String,
        greeting: j['greeting'] as String,
        oneAction: j['one_action'] as String,
        actionReason: j['action_reason'] as String,
        durationMinutes: j['duration_minutes'] as int,
        routineId: j['routine_id'] as int?,
        outfitReady: j['outfit_ready'] as bool,
      );
}

class FlowGuideResponse {
  final String phase;
  final String phaseLabel;
  final WeatherInfo? weather;
  final FlowGuideCard card;
  final String? forcedRoutine;
  final bool tomorrowOutfitSet;

  const FlowGuideResponse({
    required this.phase,
    required this.phaseLabel,
    this.weather,
    required this.card,
    this.forcedRoutine,
    required this.tomorrowOutfitSet,
  });

  factory FlowGuideResponse.fromJson(Map<String, dynamic> j) => FlowGuideResponse(
        phase: j['phase'] as String,
        phaseLabel: j['phase_label'] as String,
        weather: j['weather'] != null
            ? WeatherInfo.fromJson(j['weather'] as Map<String, dynamic>)
            : null,
        card: FlowGuideCard.fromJson(j['card'] as Map<String, dynamic>),
        forcedRoutine: j['forced_routine'] as String?,
        tomorrowOutfitSet: j['tomorrow_outfit_set'] as bool,
      );
}

File: backend/app/schemas/flow_guide.py
from pydantic import BaseModel
from enum import Enum


class TimePhase(str, Enum):
    AM = "AM"        # 06:00–11:59  황금 모닝 루틴
    PM = "PM"        # 12:00–17:59  집중·행동 시간
    EVENING = "EVENING"  # 18:00–21:59  내일 준비 + 회복
    NIGHT = "NIGHT"  # 22:00–05:59  수면 모드


class WeatherInfo(BaseModel):
    temp: float
    feels_like: float
    condition: str  # "Clear", "Clouds", "Rain", ...
    humidity: int
    city: str = "Seoul"


class FlowGuideCard(BaseModel):
    phase: TimePhase
    greeting: str
    one_action: str          # 딱 하나의 행동 지시
    action_reason: str       # 왜 지금 이걸 해야 하는지
    duration_minutes: int
    routine_id: int | None
    outfit_ready: bool       # 내일 옷이 준비됐는지


class FlowGuideResponse(BaseModel):
    phase: TimePhase
    phase_label: str         # "GOOD MORNING", "FOCUS TIME", etc.
    weather: WeatherInfo | None
    card: FlowGuideCard
    forced_routine: str | None  # 저녁엔 항상 "내일 옷 고르기"
    tomorrow_outfit_set: bool
